lib/maths/CBayesianOptimisation.cc

```diff
@@ -58,6 +59,21 @@ double stableNormPdf(double z) {
 // altogether because it is possible that the function we're interpolating has a
 // narrow deep valley that the Gaussian Process hasn't sampled.
 const double MINIMUM_KERNEL_SCALE_FOR_EXPECTATION_MAXIMISATION{1e-8};
+
+double stableProdd(double theta1, double x) {
```

Comment:
Could we maybe think of a better name for these two functions, i.e. I think that these should reflect the fact they are coming from integrals of the kernel.

Reply:
Alright, I call them `integrate1dKernel` and `integrate1dKernelProduct`. Do you think it's better?

Reply:
Sounds good.